### ISSUE TITLE
Is it possible to add summaries to an existing event file ?

### ISSUE THREAD

**Issue by cegeme** [tensorflow/tensorflow#2399]
When I continue to train my model from last interrupted training, `tf.train.SummaryWriter` create a new event file.
Could I add new summaries to my existing event file?  So I can visualize the whole learning run.

P.S. What's the right way to suspend the training binary on GPU?   We share one GPU, and I always use  "Ctrl+c" ……

**Comment by mrry** [comment]
The SummaryWriter API doesn't let you append to an existing file, but you can use TensorBoard to visualize multiple files in the same log directory.

Assigning to @danmane in case there are any other tricks you could use.

**Comment by danmane** [comment]
Correct, if you point TensorBoard to the directory containing event files, then creating a new event file within that directory will be fine, and tensorboard will combine the runs.

N.b. you should use a clean directory for each separate training job.

**Comment by yangyh1911** [comment]
see 

> TensorBoard reads data from a full directory, and organizes it into the history of a single TensorFlow execution.

at **Event Files & LogDirs: How TensorBoard loads the data** section in https://github.com/tensorflow/tensorboard/blob/master/README.md](url)

**Comment by visionscaper** [comment]
@decentralion, all, but how can I ensure that plotting of events of a certain event file, in the scalar plots, starts at the right training step? After restarting from a checkpoint I assumed I could make Tensorboard show new values starting from the training step it left off before the training restart.

Showing the horizontal axis in `relative` or `wall` mode is not a satisfactory solution to me: there is a big gap between the current run and the previous if there is a lot of time between the runs.

It seems such an obvious use case, I'm a bit surprised this hasn't been incorporated. Interested to hear what was the thinking not to enable this use case (if it's not possible).

**Comment by gzerveas** [comment]
Whenever you add summaries, you also include the global_step, like this: `summary_writer.add_summary(summary, step)`.
So it's then just a matter of passing the right global_step values when you add your summaries. You can do this for example by keeping a non-trainable Tensorflow variable called `global_step`, which is going to be checkpointed and can be restored from the checkpoint together with all other savable variables. This 0D tensor is incremented at each training step if you pass it as part of the list of tensors to `sess.run`.
Some helpful operations:
```
global_step_tensor = tf.Variable(0, trainable=False, name='global_step')
output = self.sess.run(global_step_tensor + tensors_to_run, feed_dict=feed_dict)
global_step = output[0]
```
The **real** problem is that Tensorboard doesn't let you pick your own colors for displaying curves corresponding to different event files, which means that if you restore a training session from a checkpoint, the different parts will be displayed with different colors. This can be very annoying, especially if you compare several models at the same time.
